scripts/vm_run_tests.psm1

```diff
@@ -31,44 +31,244 @@ function Invoke-CICDTestsOnVM
-              [Parameter(Mandatory=$True)] [string] $RemoteIPV6Address)
+              [Parameter(Mandatory=$True)] [string] $RemoteIPV6Address,
+              [Parameter(Mandatory=$True)] [string] $WorkingDirectory,
+              [Parameter(Mandatory=$True)] [string] $LogFileName)
             Import-Module $WorkingDirectory\common.psm1 -ArgumentList ($LogFileName) -Force -WarningAction SilentlyContinue
-            pushd $WorkingDirectory
-            Write-Log "Loading decap proggram."
-            netsh ebpf add program .\decap_permit_packet.o Pinpath=path1 2>&1 | Write-Log
-            Write-Log "Allowing XDP test app through firewall."
+            Import-Module $WorkingDirectory\run_tests.psm1 -ArgumentList ($WorkingDirectory, $LogFileName) -Force -WarningAction SilentlyContinue
+            Write-Log "Invoking $XDPTestName on $VM"
+            Invoke-XDPTest $RemoteIPV4Address $RemoteIPV6Address $XDPTestName $WorkingDirectory
+    } -ArgumentList ($VM, $XDPTestName, $RemoteIPV4Address, $RemoteIPV6Address, "C:\eBPF", ("{0}_{1}" -f $VM, $LogFileName)) -ErrorAction Stop
+}
+
+function Add-XDPTestFirewallRuleOnVM {
+    param ([parameter(Mandatory=$true)] [string] $VM,
+           [Parameter(Mandatory=$True)] [string] $LogFileName)
+
+    $TestCredential = New-Credential -Username $Admin -AdminPassword $AdminPassword
+
+    # Allow XDP Test in Firewwall on VM.
+    Invoke-Command -VMName $VM -Credential $TestCredential -ScriptBlock {
+        param([Parameter(Mandatory=$True)] [string] $VM,
+              [Parameter(Mandatory=$True)] [string] $WorkingDirectory,
+              [Parameter(Mandatory=$True)] [string] $LogFileName)
+            Import-Module $WorkingDirectory\common.psm1 -ArgumentList ($LogFileName) -Force -WarningAction SilentlyContinue
+            Write-Log "Allowing XDP test app through firewall on $VM."
             New-NetFirewallRule -DisplayName "XDP_Test" -Program "$WorkingDirectory\xdp_tests.exe" -Direction Inbound -Action Allow
-            Write-Log "Invoking xdp reflect test (IPv4)."
-            .\xdp_tests.exe xdp_reflect_test --remote-ip $RemoteIPV4Address | Write-Log
-            Write-Log "Invoking xdp reflect test (IPv6)."
-            .\xdp_tests.exe xdp_reflect_test --remote-ip $RemoteIPV6Address | Write-Log
-            Write-Log "xdp reflect tests passed." -ForegroundColor Green
-            popd
-    } -ArgumentList ("C:\eBPF", ("{0}_{1}" -f $VM1, $LogFileName), $VM1V4Address, $VM1V6Address) -ErrorAction Stop
+    } -ArgumentList ($VM, "C:\eBPF", ("{0}_{1}" -f $VM, $LogFileName)) -ErrorAction Stop
+}
+
+function Invoke-XDPTest1
+{
+    param([Parameter(Mandatory=$True)] [string] $VM1,
+          [Parameter(Mandatory=$True)] [string] $VM2,
+          [Parameter(Mandatory=$True)] [string] $VM1Int1V4Address,
+          [Parameter(Mandatory=$True)] [string] $VM1Int1V6Address,
+          [Parameter(Mandatory=$True)] [string] $VM1Int2V4Address,
+          [Parameter(Mandatory=$True)] [string] $VM1Int2V6Address,
```

Comment:
nit: "Int" looks like it may be an abbreviation for Interface and if so, better to spell it out (throughout this file) per coding conventions.  Int by itself means integer.